### ISSUE TITLE
Add `touchio` support to the M4.

### ISSUE THREAD

**Issue by tannewt** [adafruit/circuitpython#268]

**Comment by tannewt** [comment]
We're not going to block 3.0 on this. We'll add support once https://github.com/adafruit/Adafruit_FreeTouch supports it.

**Comment by kenidowa** [comment]
Will this ever happen?

**Comment by ladyada** [comment]
@kenidowa wanna help? how are you at reverse engineering ARM assembly :)

**Comment by siddacious** [comment]
I've been seriously considering giving this a try. I have no idea how successful I'll be but I may or may not have watched your qtouch videos more than a few times and sought out an alternative to IDA.

Good money says I'll fail miserably but maybe I'd at least be able to tell someone more  what _not_ to try  :)

**Comment by ladyada** [comment]
lol could look at the new free ghidra?

**Comment by kenidowa** [comment]
So, I asked about Touchio because I found it really usefully for standalone application of a TrinketMO but when I tried using it on ItsyBitsyMO on larger applications I ran into code space issues.  So, moved to ItsyBitsyM4 but then found it doesn't have Touchio.  So, after watching 2yr. old video of yours on it.  Is it that the SAMD51 touch is so different from SAMD21 or is it the amount of code space it requires in the base that's the problem.  Looking at data sheets, it isn't apparent to me that they should be that different.
